src/Tribe/Editor/Meta.php

```diff
@@ -194,7 +194,19 @@ public function meta_backwards_compatibility( $data, $post, $request ) {
 	public function update_cost( $post_data, $request ) {
 		$post_id = $request->get_param( 'id' );
 
-		\Tribe__Events__API::update_event_cost( $post_id );
+		// Fetch cost data from the request (if set).
+		$json = $request->get_json_params();
+
+		// Set an appropriate default for the cost.
+		$default = (array) tribe_get_cost( $post_id );
+
+		// If the cost is set in the submitted data, set THAT as the default.
+		if ( isset( $json['meta']['_EventCost'] ) ) {
+			$default = [ $json['meta']['_EventCost'] ];
+		}
```

Comment:
We have a utility class called `Tribe__Utils__Array` that we use to get values from an array when we don't know whether the values are set, and return a default if not. It might look something like this:

```
$json = $request->get_json_params();
$meta = Tribe__Utils__Array::get( $json, 'meta', [] );

if ( isset( $meta['_EventCost'] ) ) {
    ...
}
```

If the meta is set, it'll get the value, otherwise return an empty array.

Reply:
@paulmskim here's a video which covers all the scenarios described in the ticket. https://www.loom.com/share/74e05e0df370412988f1722ee27c7046

In regard to this question **_Towards the end of your screencast, you had the ticket, and you added the price range. When you refreshed in the FE, the range showed correctly as 50 - 100, but when you refreshed in the block editor, it only showed up as 50. Is this intended/expected?_** I spoke to @borkweb about this and he mentioned that this issue has been there for a while so I think it needs a separate bug ticket.